### DISCUSSION TITLE
[Discussion]  Implement support for other container runtimes

### DISCUSSION THREAD

**Discussion by krishjainx** [89luca89/distrobox#1888] [Ideas]
Implement container runtime support in the following order:

1. Incus system containers (LXD)
2. Bubblewrap
3. systemd-nspawn

Rationale:

1. Incus now has OCI support, making it a priority
2. A solution using Bubblewrap can be developed
3. nspawn requires dedicated images, which adds complexity to the implementation
4. This can help distrobox provide better security guarantees

Discussed with @89luca89. Self assigning

**Comment by Bubbelb** [comment]
Note: Systemd-Nspawn also supports OCI Images for a while now: https://www.phoronix.com/news/Systemd-Nspawn-OCI-Runtime

Just for clarity. Also eases installation on a lot of distributions, since systemd is already installed. Though maybe some additional packages are needed (example: in debian systemd-container needs to be installed)

**Comment by MisutaaAsriel** [comment]
This would be rather interesting and useful considering its integration with systemd. As @Bubbelb pointed out, its already on numerous systems.

**Comment by lambdadog** [comment]
I would really like to see distrobox with bubblewrap. Compared to everything distrobox can currently run on (and everything else on this list), bubblewrap would be the least of a nightmare to bootstrap in many environments, being a C project with minimal dependencies. Even lilipod depends on Go and everything that brings with it. That said, the problem with this is equally evident: being a C project with minimal dependencies, bubblewrap is *just* a sandboxing tool and distrobox clearly would prefer its container runtime be able to work with OCI images natively, including ex. downloading them.
